beluga_ros/test/test_point_cloud.cpp

```diff
@@ -0,0 +1,773 @@
+  // Fill the PointCloud2 message
+  for (unsigned i = 0; i < point_data.size() / 3; ++i) {
+    *iter_x = point_data.at(3 * i + 0);
+    *iter_y = point_data.at(3 * i + 1);
+    *iter_z = point_data.at(3 * i + 2);
+    *iter_intensity = intensity_data.at(i);
+
+    ++iter_x;
+    ++iter_y;
+    ++iter_z;
+    ++iter_intensity;
+  }
+  auto cloud = beluga_ros::PointCloud3<beluga_ros::msg::PointFieldF32>(message, origin);
+  auto map = cloud.points();
+  // Check assert
+  for (int i = 0; i < map.cols(); ++i) {
+    for (int j = 0; j < map.rows(); ++j) {
+      ASSERT_EQ(map(j, i), point_data.at(3 * i + j));
+    }
+  }
+}
+
+TEST(TestPointCloud, XYZIDoublePC) {
+  auto message = make_message();
+  const auto origin = Sophus::SE3d{};
+  message->width = 1;   // Unordered point cloud
+  message->height = 5;  // Number of points
+  // Set the point fields to x, y, z and intensity
+  int fields = 4;
+  message->fields.clear();
+  message->fields.reserve(fields);
+  // Set offset
+  int offset = 0;
+  offset = addPointField(*message, "x", 1, beluga_ros::msg::PointFieldF64, offset);
+  offset = addPointField(*message, "y", 1, beluga_ros::msg::PointFieldF64, offset);
+  offset = addPointField(*message, "z", 1, beluga_ros::msg::PointFieldF64, offset);
+  offset = addPointField(*message, "intensity", 1, beluga_ros::msg::PointFieldF64, offset);
+  // Set message params
+  message->point_step = offset;
+  message->row_step = message->width * message->point_step;
+  message->is_dense = true;
+  message->data.resize(message->point_step * message->width * message->height);
+  // Create data iterators
+  beluga_ros::msg::PointCloud2Iterator<double> iter_x(*message, "x");
+  beluga_ros::msg::PointCloud2Iterator<double> iter_y(*message, "y");
+  beluga_ros::msg::PointCloud2Iterator<double> iter_z(*message, "z");
+  beluga_ros::msg::PointCloud2Iterator<double> iter_intensity(*message, "intensity");
+  // Create some raw data for the points
+  const std::vector<double> point_data = {1.0, 2.0,  3.0,  4.0,  5.0,  6.0,  7.0, 8.0,
+                                          9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0};
+  const std::vector<double> intensity_data = {1.1, 2.2, 3.3, 4.4, 5.5};
+  // Fill the PointCloud2 message
+  for (unsigned i = 0; i < point_data.size() / 3; ++i) {
+    *iter_x = point_data.at(3 * i + 0);
+    *iter_y = point_data.at(3 * i + 1);
+    *iter_z = point_data.at(3 * i + 2);
+    *iter_intensity = intensity_data.at(i);
+
+    ++iter_x;
+    ++iter_y;
+    ++iter_z;
+    ++iter_intensity;
+  }
+  auto cloud = beluga_ros::PointCloud3<beluga_ros::msg::PointFieldF64>(message, origin);
+  auto map = cloud.points();
+  // Check assert
+  for (int i = 0; i < map.cols(); ++i) {
+    for (int j = 0; j < map.rows(); ++j) {
+      ASSERT_EQ(map(j, i), point_data.at(3 * i + j));
+    }
+  }
+}
+
+TEST(TestPointCloud, XYZPointsEmptyUnorderedPC) {
+  auto message = make_message();
```

Comment:
@pvela2017 meta: I think that having one `make_<qualifier>_message()` function for each kind of test we need would improve readability e.g. `make_empty_xyz_unordered_pointcloud_message()`, `make_xyz_ordered_pointcloud_message()`, and so on. Maybe some can reuse each other.

Reply:
Added d867fd3745f3540818b21c6dda282ca80d191689